config.m4

```diff
@@ -272,7 +273,11 @@ if test "$PHP_DDTRACE" != "no"; then
     [AC_MSG_ERROR([cannot find or include curl])])
 
   AC_CHECK_HEADER(time.h, [], [AC_MSG_ERROR([Cannot find or include time.h])])
-  PHP_SUBST(EXTRA_LDFLAGS)
+
+  dnl Only export symbols defined in ddtrace.sym, which should all be marked as
+  dnl DDTRACE_PUBLIC in their source files as well.
+  EXTRA_CFLAGS="$EXTRA_CFLAGS -fvisibility=hidden"
+  EXTRA_LDFLAGS="$EXTRA_LDFLAGS -export-symbols $ext_srcdir/ddtrace.sym"
 
```

Comment:
these lines coming after the call to substitute looks strange, even if it works ... I'm not sure how it does work in fact ...

Reply:
No problem, I can move the `PHP_SUBST` down, I'm pretty sure. The `PHP_CHECK_LIBRARY` calls just above this also append to `EXTRA_LDFLAGS`.